pkg/controller/externalmodel/reconciler_test.go

```
@@ -0,0 +1,321 @@
+
+var (
+	k8sClient client.Client
+	testEnv   *envtest.Environment
+	ctx       context.Context
+	cancel    context.CancelFunc
+)
+
+func TestMain(m *testing.M) {
+	ctrl.SetLogger(zap.New(zap.WriteTo(os.Stderr), zap.UseDevMode(true)))
+
+	ctx, cancel = context.WithCancel(context.Background())
```

golangci-lint (2.12.1)

[error] nested context in function literal

(fatcontext)